### DISCUSSION TITLE
Estimated export size disparity?

### DISCUSSION THREAD

**Discussion by chrisdrackett** [ReagentX/imessage-exporter#369] [Q&A] [answered]
When I went to run the export it said their wasn't room (export would take up 74GB, but only 21GB was free). However my HD had 234GB free at the time and the final export (done with `--ignore-disk-warning`) ended up only being 23GB.

My only real question apart from reporting is if I should be concerned that the estimate was 74GB and the actual export was only 23GB. Is there a chance I'm missing data in the export?

**Comment by ReagentX** [comment, marked as answer]
The estimated file size is based on the sum of the file sizes referenced in the table:

https://github.com/ReagentX/imessage-exporter/blob/a05922a408efefbe166a507a3d60c41440b7a98c/imessage-database/src/tables/attachment.rs#L243-L277

To check if you have missing data, i.e. attachments not synced in from iCloud, use the `-d` flag to emit [diagnostics](https://github.com/ReagentX/imessage-exporter/blob/develop/docs/diagnostics.md).

**Reply by ReagentX** [reply]
I would be more concerned that the free space is getting reported incorrectly; internally `imessage-exporter` uses `statvfs` which is a `libc` call; it should be accurate.

**Comment by chrisdrackett** [comment]
![TPAbbUTvGZYnRNySvxS2TRZiFOIBR4P7ZBhp07Qb](https://github.com/user-attachments/assets/94c7873e-d13c-44c6-8c99-debca0b93426)

Yeah, the size thing is super odd. I actually just removed a bunch of things and it is still way off.

**Comment by chrisdrackett** [comment]
When I run with `-d` I get the following, which seems odd because without the `-d` it runs.

```
Invalid options!
Diagnostics are enabled; copy-method is disallowed
```

The command I'm using:

`imessage-exporter -f html -c efficient -o $OUTPUT_DIR -m chrisdrackett -p $DB -r $AT -d --ignore-disk-warning`

**Reply by ReagentX** [reply]
The reason it does not run is explained in the message it emitted:

```
Diagnostics are enabled; copy-method is disallowed
```

See the [docs](https://github.com/ReagentX/imessage-exporter/blob/develop/imessage-exporter/README.md#how-to-use) for more info.
